src/main/java/org/apache/commons/compress/archivers/ArchiveStreamFactory.java

```diff
@@ -268,7 +269,16 @@ public static String detect(final InputStream in) throws ArchiveException {
             try {
                 tais = new TarArchiveInputStream(new ByteArrayInputStream(tarHeader));
                 // COMPRESS-191 - verify the header checksum
-                if (tais.getNextTarEntry().isCheckSumOK()) {
+                // COMPRESS-644 - do not allow zero byte file entries
+                TarArchiveEntry tae = tais.getNextTarEntry();
+                //try to find the first non-directory entry within the first 10 entries.
+                int max = 10;
```

Comment:
What makes 10 special? Needs a comment or a better a Javadoc comment for a new constant.

Reply:
Completely arbitrary.  I improved documentation.  I'm completely open to the idea of dropping this attempt to iterate through the first x directory entries.